### PR TITLE
Support array comparison

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
fixes https://github.com/lcompilers/lpython/issues/2135

### CODE REVIEW

File: tests/reference/asr-arrays_02-da94458.stderr [hunk 1]
```diff
@@ -0,0 +1,5 @@
+semantic error: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```
Comment (certik) [inline tests/reference/asr-arrays_02-da94458.stderr:1]: How about this:
```suggestion
semantic error: The truth value of an array with more than one element is ambiguous. Use any() or all()
```

File: tests/reference/asr-arrays_02-da94458.stderr [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  --> tests/errors/arrays_02.py:28:8
+   |
+28 | assert r1.a == t1.a
+   |        ^^^^^^^^^^^^
```
Comment (certik) [inline tests/reference/asr-arrays_02-da94458.stderr:5]: And this:
```
help: use any(r1.a == t1.a) or all(r1.a == t1.a)
```

**Review by certik — APPROVED** [review]
I think that looks good.

You can merge this, and in the next PR you can try to improve the error message to be more helpful, I provided some ideas.

File: tests/reference/asr-arrays_02-da94458.stderr (hunk 2 above)
Comment (Shaikh-Ubaid) [reply tests/reference/asr-arrays_02-da94458.stderr:5]: >help: use any(r1.a == t1.a) or all(r1.a == t1.a)

Here, we do not have access to the string `r1.a == t1.a`. When printing diagnostics the input file is read (from disk) and the error line/string is obtained form the read source input.

Shall we do the following?
```
help: use any() or all()
```
Or shall we keep the message as it is currently in the main branch?

Comment (certik) [reply tests/reference/asr-arrays_02-da94458.stderr:5]: We have access to AST and partial ASR, so we should be able to construct `t1`, `r1` and `a`. And we know it's a struct access and we know it's a comparison. So you use this information to create the text `any(r1.a == t1.a)`.

Comment (certik) [reply tests/reference/asr-arrays_02-da94458.stderr:5]: I made this https://github.com/lcompilers/lpython/issues/2159, so that we don't forget.

**Pull request merged**